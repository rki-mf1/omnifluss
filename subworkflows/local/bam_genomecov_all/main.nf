include { BEDTOOLS_GENOMECOV } from '../../../modules/nf-core/bedtools/genomecov/main' //get_read_cov.smk

workflow BAM_GENOMECOV_ALL {

    take:
    ch_mapping                  // channel: [ val(meta), bam]
    sizes                       // File
    extension                   // String
    sort                        // Bool

    main:
    ch_versions = Channel.empty()

    //get_read_cov.smk
    ch_mapping_bedtools = ch_mapping 
        | map {meta, bam -> [meta, bam, 1]} //setting scale factor to 1 disables it.

    BEDTOOLS_GENOMECOV(
        ch_mapping_bedtools, 
        sizes,      
        extension,
        sort
    )
    ch_cov_bedtools = BEDTOOLS_GENOMECOV.out.genomecov
    ch_versions = ch_versions.mix(BEDTOOLS_GENOMECOV.out.versions)

    emit:
    bedtools_cov         = ch_cov_bedtools
    versions             = ch_versions
    
}