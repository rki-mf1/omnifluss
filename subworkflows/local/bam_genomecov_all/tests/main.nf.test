nextflow_workflow {
    name "Test Subworkflow BAM_GENOMECOV_ALL"
    script "../main.nf"
    // Incude a nextflow.config for the test if needed
    config "subworkflows/local/bam_genomecov_all/tests/nextflow.config"
    workflow "BAM_GENOMECOV_ALL"

    //  Add tags for the subworkflow
    tag "subworkflows"
    tag "subworkflows/local"
    tag "subworkflows/bam_genomecov_all"
    // Add tags for all modules used within this subworkflow
    tag "bedtools"

    test("sarscov2  - stub") {

        options '-stub'

        when {
            workflow {
                """
                input[0] = Channel.of([
                [id:'test'],
                file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.bam', checkIfExists: true)
                ])
                input[1] = []
                input[2] = 'test.tsv'
                input[3] = true
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

    test("sarscov2") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                [id:'test'],
                file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.bam', checkIfExists: true)
                ])
                input[1] = []
                input[2] = 'test.tsv'
                input[3] = true
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }


}