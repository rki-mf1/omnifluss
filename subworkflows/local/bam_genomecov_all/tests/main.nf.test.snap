{
    "sarscov2  - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,338f4892744593cda3dba7e072416f9b"
                ],
                "bedtools_cov": [
                    [
                        {
                            "id": "test"
                        },
                        "test.test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,338f4892744593cda3dba7e072416f9b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2025-03-27T08:03:56.878193748"
    },
    "sarscov2": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.test.tsv:md5,122381298ce5c7d5ccf3b570d4b0f2b1"
                    ]
                ],
                "1": [
                    "versions.yml:md5,338f4892744593cda3dba7e072416f9b"
                ],
                "bedtools_cov": [
                    [
                        {
                            "id": "test"
                        },
                        "test.test.tsv:md5,122381298ce5c7d5ccf3b570d4b0f2b1"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,338f4892744593cda3dba7e072416f9b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2025-03-27T08:04:32.899218466"
    }
}